src/status.rs

```diff
@@ -103,13 +104,128 @@ fn estimated_total_task_time(
 	}
 }
 
+#[derive(Debug, Clone, PartialEq)]
+pub enum ProgressFormat {
+	None,
+	Text,
+	ASCIIBar,
+	HighResBar,
+}
+
+#[derive(Debug)]
+pub struct ParseProgressFormatError {
+	value: String,
+}
+
+impl core::str::FromStr for ProgressFormat {
+	type Err = ParseProgressFormatError;
+	fn from_str(s: &str) -> Result<Self, ParseProgressFormatError> {
+		let s = s.to_lowercase();
+		if s == "none" {
+			Ok(ProgressFormat::None)
+		} else if s == "text" {
+			Ok(ProgressFormat::Text)
+		} else if s == "ascii" {
+			Ok(ProgressFormat::ASCIIBar)
+		} else if s == "highres" {
+			Ok(ProgressFormat::HighResBar)
+		} else {
+			Err(ParseProgressFormatError { value: s })
+		}
+	}
+}
+
+impl fmt::Display for ParseProgressFormatError {
+	fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+		write!(f, "{}", self.value)
+	}
+}
+
+fn render_progress_bar(width: usize, percentage: f32, text: &String, highres: bool) -> String {
+	while text.len() > width {
```

Comment:
A 'loop' with a `return` statement?

Reply:
Whoops, that was an `if` before and I made it a loop trying to shorten the text, but aborted that attempt and forgot to change it back